lib/user-profile-manager/user-profile-manager.js

```diff
@@ -103,12 +108,65 @@ UserAttributeManager.prototype.getAllAttributes = function (accessToken) {
 	return deferred.promise;
 };
 
-function handleRequest(accessToken, attributeValue, method, url, action, deferred) {
-	
+/**
+ * Retrieves user info using the provided accessToken
+ *
+ * @param accessToken {string} - the accessToken string used for authorization
+ * @param identityToken {string|undefined} - an optional identity token. If provided, will be used to validate UserInfo response
+ * @returns Promise<JSON> - the user info in json format
+ */
+UserProfileManager.prototype.getUserInfo = function (accessToken, identityToken) {
+	const deferred = Q.defer();
+	const internalDeferred = Q.defer();
+
+	if (!_.isString(accessToken)) {
+		logger.error("getUserinfo invalid invocation parameter");
+		return Q.reject();
+	}
+
+	logger.debug("Getting userinfo");
+	const serverUrl = this.oauthServerUrl + USERINFO_ENDPOINT;
```

Comment:
save this endpoint as global, it redundant to create it every API call if it does not change

Reply:
@rotembr The other apis also generate their request urls and use it primarily for testing. Users are able to modify the oauthServerUrl / userProfileServiceUrl, which means these would have to be dynamic in order to support changes. If you thinks its better to lock a UserProfileManager instance to the initial configuration, we should probably follow the same model across all the apis? What do you think?

Reply:
I see, lets leave it this way for now.